test/integration/simple-token.js

```diff
@@ -1,91 +1,87 @@
 const solc = require('solc')
 const TokenTracker = require('../../lib')
 const BN = require('ethjs').BN
 
 const Eth = require('ethjs-query')
 const EthContract = require('ethjs-contract')
-const eth = new Eth(provider)
-const contract = new EthContract(eth)
 
 const source = fs.readFileSync(path.resolve(__dirname, '..', 'contracts/Token.sol')).toString();
 const compiled = solc.compile(source, 1)
 const SimpleTokenDeployer = compiled.contracts[':SimpleToken']
 
-let addresses = []
-let token, tokenAddress, tracked
-
-test('testrpc has addresses', function (t) {
-  eth.accounts()
-  .then((accounts) => {
-    addresses = accounts
-    t.ok(accounts, 'loaded accounts')
-    t.end()
-  })
-})
-
+let tracked
 const qty = '100000000000000000000' // 100 x 10 ^ 18
 const less = '10000000000000000000' // 100 x 10 ^ 17
-test('StandardToken publishing token & checking balance', function (t) {
-  const abi = JSON.parse(SimpleTokenDeployer.interface)
+
+async function setupSimpleTokenEnvironment () {
+  const provider = TestRPC.provider()
+  const eth = new Eth(provider)
+
+  const addresses = await eth.accounts()
+  assert(addresses.length > 0, 'test network should be initialized with accounts')
+
   const owner = addresses[0]
+  const contract = new EthContract(eth)
+  const abi = JSON.parse(SimpleTokenDeployer.interface)
   const StandardToken = contract(abi, SimpleTokenDeployer.bytecode, {
     from: owner,
     gas: '3000000',
     gasPrice: '30000',
   })
-  StandardToken.new(qty)
-  .then((txHash) => {
-    t.ok(txHash, 'publishes a txHash')
 
-    return new Promise((res, rej) => {
-      setTimeout(() => res(txHash), 300)
-    })
-  })
-  .then((txHash) => {
-    return eth.getTransactionReceipt(txHash)
-  })
-  .then((receipt) => {
-    const addr = receipt.contractAddress
-    t.ok(addr, 'should have an address')
-    tokenAddress = addr
-    token = StandardToken.at(addr)
-    return token.balanceOf(owner)
-  })
-  .then((res) => {
-    const balance = res[0]
-    t.equal(balance.toString(10), qty, 'owner should have all')
-    t.end()
-  })
-  .catch((reason) => {
-    t.notOk(reason, 'should not throw error')
-    t.end()
-  })
-})
+  const txHash = await StandardToken.new(qty)
+  assert.ok(txHash, 'should have published the token and returned a transaction hash')
+
+  await new Promise((resolve) => setTimeout(resolve, 300))
+  const receipt = await eth.getTransactionReceipt(txHash)
+
+  const tokenAddress = receipt.contractAddress
+  assert.ok(tokenAddress, 'should have a token address')
+
+  const token = StandardToken.at(tokenAddress)
+  const result = await token.balanceOf(owner)
+  const balance = result[0]
+  assert.equal(balance.toString(10), qty, 'owner should have all')
+
+  return { addresses, eth, provider, token, tokenAddress }
+}
 
 test('StandardToken balances are tracked', function (t) {
+  let addresses
+  let eth
+  let token
+  let tokenAddress
+  let tokenTracker
+  setupSimpleTokenEnvironment()
+  .then((environment) => {
```

Comment:
With certain things like indentation and the use of `.then` rather than `async/await`, I kept with the existing conventions to minimize the number of changes.